OsmAnd/src/net/osmand/plus/views/layers/geometry/GpxGeometryWay.java

```diff
@@ -192,4 +208,25 @@ public void clearWay() {
 			super.clearWay();
 		}
 	}
+
+	@Override
+	public void calculatePath(@NonNull List<Integer> indexes, @NonNull List<Integer> xs, @NonNull List<Integer> ys, @Nullable List<GeometryWayStyle<?>> styles, @NonNull List<GeometryWayDrawer.DrawPathData31> pathsData) {
+		super.calculatePath(indexes, xs, ys, styles, pathsData);
+		for (int i = 0; i < pathsData.size(); i++) {
+			GeometryWayDrawer.DrawPathData31 drawPathData = pathsData.get(i);
```

Comment:
I am not sure if it is ok to calculate points' height here, or if we should calculate them similarly to cache arrays in GeometryWay